src/main.cc

```diff
@@ -0,0 +1,88 @@
+#ifdef ONPC
+#include "../algo/debug.h"
+#else
+#define debug(...) 42
+#endif
+#include <cassert>
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+template<typename T>
+using Pair = std::pair<T, T>;
+
+template<typename T, typename Functor>
+struct FunctorQueue {
+private:
+  std::vector<Pair<T>> stack_first, stack_second;
+  Functor func;
+
+public:
+  void insert(T el);
+  void pop();
+  T get();
+
+  FunctorQueue(Functor func = Functor{}) : func(func) {}
+};
+
+template<typename T, typename Functor>
+void FunctorQueue<T, Functor>::insert(T el) {
```

Comment:
Maybe I'm already nitpicking, but better not to shrink variable names, `el -> element` it is not that hard, you write code once, but it will be read multiple times.